components/cmake/build_model.cmake

```diff
@@ -74,12 +74,17 @@ function(build_model COMP_CLASS COMP_NAME)
         # CUDA_FLAGS is set through Macros.cmake
         # For instance: cime_config/machines/cmake_macros/gnugpu_summit.cmake
         set(YAKL_CUDA_FLAGS "${CPPDEFS} ${CUDA_FLAGS}")
-	elseif (USE_HIP)
-		set(YAKL_ARCH "HIP")
+      elseif (USE_HIP)
+	set(YAKL_ARCH "HIP")
         # HIP_FLAGS are set through Macros.cmake
         # For instance: cime_config/machines/cmake_macros/crayclanggpu_frontier.cmake
-		set(YAKL_HIP_FLAGS "${CPPDEFS} ${HIP_FLAGS}")
-    else()
+	set(YAKL_HIP_FLAGS "${CPPDEFS} ${HIP_FLAGS}")
+      elseif (USE_SYCL)
+        set(YAKL_ARCH "SYCL")
+        # SYCL_FLAGS is set through Macros.cmake
+	# For instance: cime_config/machines/cmake_macros/oneapi-ifxgpu_sunspot.cmake
+        set(YAKL_SYCL_FLAGS "${CPPDEFS} ${SYCL_FLAGS}")	
```

Comment:
I had to add CPPDEFS to YAKL flags for HIP to make passing MPI include paths etc. easier. 
However, Nvidia/CUDA doesn't like that apparently. So, I presume Intel would also be okay with CPPDEFS added?

cc @mrnorman